File: backend/src/main/java/com/shoppingmall/entity/OrderMessage.java
package com.shoppingmall.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

/**
 * 订单问题/消息实体类
 *
 * @author ShoppingMall Team
 * @date 2025-12-14
 */
@Data
@TableName("order_message")
public class OrderMessage {

    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 订单号
     */
    private String orderNo;

    /**
     * 用户ID
     */
    private Long userId;

    /**
     * 消息类型：1-我已付款，2-我有问题
     */
    private Integer messageType;

    /**
     * 问题标题（我有问题时必填）
     */
    private String title;

    /**
     * 问题内容（我有问题时必填）
     */
    private String content;

    /**
     * 付款金额（我已付款时必填）
     */
    private BigDecimal paymentAmount;

    /**
     * 付款方式（我已付款时必填）
     */
    private String paymentMethod;

    /**
     * 付款日期（我已付款时必填）
     */
    private LocalDate paymentDate;

    /**
     * 付款时间（我已付款时必填）
     */
    private LocalTime paymentTime;

    /**
     * 备注
     */
    private String remarks;

    /**
     * 处理状态：0-待处理，1-处理中，2-已处理，3-已关闭
     */
    private Integer status;

    /**
     * 处理人ID（管理员）
     */
    private Long handlerId;

    /**
     * 处理人姓名
     */
    private String handlerName;

    /**
     * 处理时间
     */
    private LocalDateTime handleTime;

    /**
     * 处理备注
     */
    private String handleRemark;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: admin-frontend/src/api/admin/orderMessage.ts
/**
 * 订单问题/消息管理API
 */

import request from '@/utils/request'

/**
 * 订单问题/消息VO
 */
export interface OrderMessageVO {
  id: number
  orderNo: string
  userId: number
  username?: string
  realName?: string
  messageType: number
  messageTypeText: string
  title?: string
  content?: string
  paymentAmount?: number
  paymentMethod?: string
  paymentDate?: string
  paymentTime?: string
  remarks?: string
  status: number
  statusText: string
  handlerId?: number
  handlerName?: string
  handleTime?: string
  handleRemark?: string
  createTime: string
  updateTime: string
}

/**
 * 订单问题查询DTO
 */
export interface OrderMessageQueryDTO {
  orderNo?: string
  userId?: number
  messageType?: number
  status?: number
  startDate?: string
  endDate?: string
  current?: number
  size?: number
}

/**
 * 订单问题处理DTO
 */
export interface OrderMessageHandleDTO {
  id: number
  status: number
  handleRemark?: string
}

/**
 * 分页响应
 */
export interface PageResponse<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

/**
 * 分页查询订单问题列表
 */
export const getOrderMessagePage = (
  params: OrderMessageQueryDTO
): Promise<PageResponse<OrderMessageVO>> => {
  return request.get('/api/admin/order-messages/page', { params })
}

/**
 * 根据ID获取订单问题详情
 */
export const getOrderMessageById = (id: number): Promise<OrderMessageVO> => {
  return request.get(`/api/admin/order-messages/${id}`)
}

/**
 * 处理订单问题
 */
export const handleOrderMessage = (data: OrderMessageHandleDTO): Promise<void> => {
  return request.put('/api/admin/order-messages/handle', data)
}











































